### PR TITLE
nrf_security: cracen: Add helper functions for common bit manipulation

### PR DESCRIPTION
**Pull request by nordic-mare**
Add helper functions for the common bit manipulation operations described in RFC 7748.

### CODE REVIEW

File: subsys/nrf_security/src/drivers/cracen/silexpk/src/ed25519.c [hunk 1]
```diff
@@ -138,7 +139,7 @@ static inline int ed25519_decode_pt_x(const struct sx_ed25519_pt *pt)
 static inline void ed25519_pt_write_y(const struct sx_ed25519_pt *pt, char *ay)
 {
 	sx_wrpkmem(ay, pt->encoded, SX_ED25519_PT_SZ);
-	sx_wrpkmem_byte(&ay[SX_ED25519_PT_SZ - 1], pt->encoded[SX_ED25519_PT_SZ - 1] & 0x7f);
```
Comment (vili-nordic) [inline subsys/nrf_security/src/drivers/cracen/silexpk/src/ed25519.c:141]: I don't think it's safe to not use sx_wrpkmem_byte here

File: subsys/nrf_security/src/drivers/cracen/silexpk/src/ed448.c [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #include <silexpk/core.h>
 #include <silexpk/ed448.h>
 #include <silexpk/ec_curves.h>
+#include <cracen/math_helpers.h>
```
Comment (vili-nordic) [inline subsys/nrf_security/src/drivers/cracen/silexpk/src/ed448.c:12]: did you intend to use the helper functions in this file as well?

File: subsys/nrf_security/src/drivers/cracen/common/include/cracen/math_helpers.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * @brief Decode a byte array containing a coordinate in place.
+ *
+ * @note Corresponds to the decodeUCoordinate function in RFC 7748.
+ *
+ * @param[in] u    Byte array containing a coordinate.
+ * @param[in] size Size of the coordinate to be decoded in bytes.
+ */
+void decode_u_coordinate(uint8_t *u, uint8_t size);
```
Comment (vili-nordic) [inline subsys/nrf_security/src/drivers/cracen/common/include/cracen/math_helpers.h:19]: the implementation here only does decodeUCoordinate for 25519. For 448 decodeUCoordinate does not nothing. Maybe rename the function to ...25519 and remove the size parameter, or update the implementation to match the RFC? I see the RFC takes the size in bits.

**Review by mswarowsky — COMMENTED** [review]
Nit, maybe call it `ec_helpers` as RFC 7748 is for elliptic curves only

File: subsys/nrf_security/src/drivers/cracen/silexpk/src/ed448.c (hunk 2 above)
Comment (nordic-mare) [reply subsys/nrf_security/src/drivers/cracen/silexpk/src/ed448.c]: No, removed it

File: subsys/nrf_security/src/drivers/cracen/silexpk/src/ed25519.c (hunk 1 above)
Comment (nordic-mare) [reply subsys/nrf_security/src/drivers/cracen/silexpk/src/ed25519.c]: Reverted to use sx_wrpkmem_byte

File: subsys/nrf_security/src/drivers/cracen/common/include/cracen/ec_helpers.h [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+#pragma once
+
+#include <stdint.h>
+
+/**
+ * @brief Decode a byte array containing a coordinate in place.
+ *
+ * @note Corresponds to the decodeUCoordinate function in RFC 7748 for Curve25519.
+ *
+ * @param[in] u    Byte array containing a coordinate of size 32 bytes.
```
Comment (vili-nordic) [inline subsys/nrf_security/src/drivers/cracen/common/include/cracen/ec_helpers.h:16]: in,out

**Review by vili-nordic — APPROVED** [review]

**Pull request merged**